tests/integration/test_application.py

```diff
@@ -241,6 +241,20 @@ async def test_upgrade_charm_resource_same_rev_no_update(event_loop):
         assert ress['policyd-override'].fingerprint == ress2['policyd-override'].fingerprint
 
 
+@base.bootstrapped
+@pytest.mark.asyncio
+async def test_refresh_charmhub_to_local(event_loop):
+    charm_path = INTEGRATION_TEST_DIR / 'charm'
+    async with base.CleanModel() as model:
+        app = await model.deploy('ubuntu', application_name='ubu-path')
+        await app.refresh(path=str(charm_path))
+        assert app.data['charm-url'].startswith('local:')
+
+        app = await model.deploy('ubuntu', application_name='ubu-switch')
+        await app.refresh(switch=str(charm_path))
+        assert app.data['charm-url'].startswith('local:')
+
+
```

Comment:
Please add a test case (here or in a separate one) where we have both `path` and `switch` at the same time to clarify behavior.

Reply:
Fair, I can assert it raises an error. This feels like a unit-test job though